### PR TITLE
Editable Student Names

### PR DESCRIPTION
**Pull request by johnjeng**
#### Step 1:
step 1

#### Step 2: Save disabled when nothing has changed
step 2

#### Step 3: Tabbing from first name to last name works
step 3

#### Step 4: Hitting the `enter` key works, as does clicking `Save`
final

#### Changed names are used elsewhere for just the course
elsewhere

## Test
Follow above steps.
Additionally, confirm the following:
1. Autofocuses the first name field
1. Hitting cancel resets the name
2. Hitting `Esc` also cancels
2. Editing after hitting cancel doesn't leave the previously canceled names in the input boxes
3. You have to cancel or save to close the more dialogue
4. The new name appears elsewhere once saved
5. The new name stays around after closing course file or navigating away.

### CODE REVIEW

**Comment by williamstein** [comment]
It's a little weird if the student hasn't signed up yet, since you edit their name, but only their email address is shown.  It would make sense to show their name now that you know it.

File: src/smc-webapp/course/main.cjsx [hunk 1]
```diff
@@ -1549,20 +1558,30 @@ init_redux = (course_filename, redux, course_project_id) ->
...
+        # Uses an instructor given name if it exists
         get_student_name: (student, include_email=false) =>
             student = @get_student(student)
             if not student?
                 return 'student'
             email = student.get('email_address')
-            name = user_store.get_name(student.get('account_id'))
-            n = name ? email ? 'student'
+            account_id = student.get('account_id')
+            first_name = student.get('first_name') ? user_store.get_first_name(account_id)
```
Comment (haraldschilly) [inline src/smc-webapp/course/main.cjsx]: @timothyclemansinsea see the line above, there is something going on how the first and last names of student's are derived.

**Comment by johnjeng** [comment]
- Removed ability to edit a student who has yet to sign up
- Usernames now display the name the account owner has given them if it's different from the system name.
- A tool tip has been added for clarity
- Changes in the account name or course name now correctly propagate.

- Made an issue for invited students.

**Comment by williamstein** [comment]
this seems great to me!!   I've also added this followup: https://github.com/sagemathinc/smc/issues/1669

**Pull request merged**